src/Files.App/Filesystem/Security/AccessControlList.cs

```diff
@@ -0,0 +1,178 @@
+{
+	/// <summary>
+	/// Represents an ACL.
+	/// </summary>
+	public class AccessControlList
+	{
+		public AccessControlList(string path, bool isFolder)
+		{
+			_path = path;
+			_isFolder = isFolder;
+
+			AccessControlEntryPrimitiveMappings = new();
+		}
+
+		public string? OwnerSID { get; private set; }
+
+		public Principal Owner { get; private set; }
+
+		public bool IsAccessControlListProtected { get; private set; }
+
+		public bool CanReadAccessControl { get; private set; }
+
+		public ObservableCollection<AccessControlEntry> AccessControlEntries { get; private set; }
+
+		public List<AccessControlEntryPrimitiveMapping> AccessControlEntryPrimitiveMappings { get; private set; }
+
+		private readonly string _path;
+
+		private readonly bool _isFolder;
+
+		public bool SetAccessControl()
+		{
+			if (GetAccessControl(_path, _isFolder, out var fileSystemSecurity))
+			{
+				try
+				{
+					var accessRules = fileSystemSecurity.GetAccessRules(true, true, typeof(SecurityIdentifier));
+
+					// Remove all existing rules
+					foreach (var existingRule in accessRules.Cast<FileSystemAccessRule>().Where(x => !x.IsInherited))
+						fileSystemSecurity.RemoveAccessRule(existingRule);
+
+					// Bring back removed rules, including changed rules
+					foreach (var rule in AccessControlEntryPrimitiveMappings.Where(x => !x.IsInherited))
+						fileSystemSecurity.AddAccessRule(rule.ToFileSystemAccessRule());
+
+					// Save changes
+					if (_isFolder)
+						FileSystemAclExtensions.SetAccessControl(new DirectoryInfo(_path), (DirectorySecurity)fileSystemSecurity);
+					else
+						FileSystemAclExtensions.SetAccessControl(new FileInfo(_path), (FileSecurity)fileSystemSecurity);
+
+					return true;
+				}
+				catch (UnauthorizedAccessException) { }
+				catch (Exception) { }
+			}
+
+			return false;
+		}
+
+		public static AccessControlList FromPath(string path, bool isFolder)
+		{
+			// Get ACL
+			var result = GetAccessControl(path, isFolder, out var fileSystemSecurity);
+			if (result)
+			{
+				var ownerSid = fileSystemSecurity.GetOwner(typeof(SecurityIdentifier)).Value;
+
+				var acl = new AccessControlList(path, isFolder)
+				{
+					IsAccessControlListProtected = fileSystemSecurity.AreAccessRulesProtected,
+					OwnerSID = ownerSid,
+					Owner = Principal.FromSid(ownerSid),
+					CanReadAccessControl = true,
+				};
+
+				// Get all ACEs and map them with primitive class
+				var accessRules = fileSystemSecurity.GetAccessRules(true, true, typeof(SecurityIdentifier));
+				var rules = new List<AccessControlEntryPrimitiveMapping>();
+				foreach (var accessRule in accessRules)
+					rules.Add(AccessControlEntryPrimitiveMapping.FromFileSystemAccessRule((FileSystemAccessRule)accessRule));
+
+				acl.AccessControlEntryPrimitiveMappings.AddRange(rules);
+				acl.AccessControlEntries = new(rules.Select(x => new AccessControlEntry(x, isFolder)));
+
+				return acl;
+			}
+			else
+			{
+				return new(path, isFolder);
+			}
```

Comment:
`Else` keyword is redundant here.

Reply:
Not redundant.

Reply:
It actually is, there is a `return` inside the `if`, and no other instructions after the `else`.
But it's really small a detail.